interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -190,4 +196,22 @@ object TapirAdapter {
 
   def isFtv1Header(r: Header): Boolean =
     r.name == GraphQLRequest.`apollo-federation-include-trace` && r.value == GraphQLRequest.ftv1
+
+  private def computeCacheDirective(extensions: Option[ResponseValue.ObjectValue]): Option[String] =
+    extensions
+      .flatMap(_.fields.collectFirst { case (Caching.DirectiveName, ResponseValue.ObjectValue(fields)) =>
+        fields.collectFirst { case ("policy", ResponseValue.ObjectValue(fields)) =>
+          val maxAge = fields.collectFirst { case (Caching.MaxAgeName, v: IntValue) =>
+            CacheDirective.MaxAge(FiniteDuration(v.toInt, TimeUnit.SECONDS))
+          }
+          val scope  = fields.collectFirst {
+            case (Caching.ScopeName, Value.EnumValue("PRIVATE")) => CacheDirective.Private
+            case (Caching.ScopeName, Value.EnumValue("PUBLIC"))  => CacheDirective.Public
+          }.getOrElse(CacheDirective.Public)
```

Comment:
This feels like a dangerous default value. Perhaps something like this is a bit safer?

```scala
maxAge match {
  case None    => CacheDirective.NoCache
  case Some(_) => CacheDirective.Private
}
```

Reply:
I think I captured this behavior